ts/src/bitmex.ts

```diff
@@ -261,6 +263,172 @@ export default class bitmex extends Exchange {
+                    'deposit': isDepositEnabled,
+                    'withdraw': isWithdrawEnabled,
+                    'fee': withdrawalFee,
+                    'precision': undefined,
+                    'limits': {
+                        'withdraw': {
+                            'min': undefined,
+                            'max': undefined,
+                        },
+                        'deposit': {
+                            'min': undefined,
+                            'max': undefined,
+                        },
+                    },
+                };
+            }
+            const currencyEnabled = this.safeValue (currency, 'enabled');
+            const currencyActive = currencyEnabled || (depositEnabled || withdrawEnabled);
+            const minWithdrawalString = this.safeString (currency, 'minWithdrawalAmount');
+            const minWithdrawal = this.parseNumber (Precise.stringMul (minWithdrawalString, precisionString));
+            const maxWithdrawalString = this.safeString (currency, 'maxWithdrawalAmount');
+            const maxWithdrawal = this.parseNumber (Precise.stringMul (maxWithdrawalString, precisionString));
+            const minDepositString = this.safeString (currency, 'minDepositAmount');
+            const minDeposit = this.parseNumber (Precise.stringMul (minDepositString, precisionString));
+            result[code] = {
+                'id': id,
+                'code': code,
+                'info': currency,
+                'name': name,
+                'active': currencyActive,
+                'deposit': depositEnabled,
+                'withdraw': withdrawEnabled,
+                'fee': undefined,
+                'precision': precision,
+                'limits': {
+                    'amount': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'withdraw': {
+                        'min': minWithdrawal,
+                        'max': maxWithdrawal,
+                    },
+                    'deposit': {
+                        'min': minDeposit,
+                        'max': undefined,
+                    },
+                },
+                'networks': networks,
+            };
+        }
+        return result;
+    }
+
+    amountToCurrencyPrecision (code, amount) {
+        const currency = this.currency (code);
+        const precision = this.safeString (currency, 'precision');
+        const amountString = this.numberToString (amount);
+        const finalAmount = Precise.stringDiv (amountString, precision);
+        return this.parseNumber (finalAmount);
+    }
+
+    convertToRealAmount (code, amount) {
+        const currency = this.currency (code);
+        const precision = this.safeString (currency, 'precision');
+        const amountString = this.numberToString (amount);
+        const finalAmount = Precise.stringMul (amountString, precision);
```

Comment:
```suggestion
        const finalAmount = Precise.stringMul (amount, precision);
```
`amount` argument is already provided as 'string' from all cases.

Reply:
Not necessarily, I think it's safer to always make sure types match

Reply:
I am not sure, why to add extra manipulations while the method already receives strings. otherwise, you might add extra lines here:  https://github.com/ccxt/ccxt/blob/master/python/ccxt/base/decimal_to_precision.py#L168 
so it wouldnt reparse into string->decimal->string.
somewhat like:
```
def number_to_string(x):
    # avoids scientific notation for too large and too small numbers
    if x is None:
        return None
    if isinstance(x, str)
        return x
```

Reply:
here - https://github.com/ccxt/ccxt/pull/18109

Reply:
I think we can keep as is right now and address this issue separately if needed